src/frontend/src/flows/authenticate/index.ts

```diff
@@ -0,0 +1,356 @@
+import { html, render } from "lit-html";
+import { editIcon } from "../../components/icons";
+import { navbar } from "../../components/navbar";
+import { footer } from "../../components/footer";
+import {
+  getUserNumber,
+  parseUserNumber,
+  setUserNumber,
+} from "../../utils/userNumber";
+import { withLoader } from "../../components/loader";
+import { IIConnection } from "../../utils/iiConnection";
+import {
+  apiResultToLoginFlowResult,
+  LoginFlowSuccess,
+} from "../login/flowResult";
+import { displayError } from "../../components/displayError";
+import { useRecovery } from "../recovery/useRecovery";
+import waitForAuthRequest, { AuthContext } from "./postMessageInterface";
+import { toggleErrorMessage } from "../../utils/errorHelper";
+import { fetchDelegation } from "./fetchDelegation";
+import { registerIfAllowed } from "../../utils/registerAllowedCheck";
+
+const pageContent = (hostName: string, userNumber?: bigint) => html` <style>
+    .anchorText {
+      font-size: 1.5rem;
+    }
+
+    #userNumberInput {
+      margin: 0;
+      text-align: center;
+      width: 100%;
+      box-sizing: border-box;
+      padding: 0.5rem 2rem;
+    }
+
+    #userNumberInput:focus {
+      box-sizing: border-box;
+      border-style: double;
+      border-width: 2px;
+      border-radius: 4px;
+      border-image-slice: 1;
+      outline: none;
+      border-image-source: linear-gradient(
+        270.05deg,
+        #29abe2 10.78%,
+        #522785 22.2%,
+        #ed1e79 42.46%,
+        #f15a24 59.41%,
+        #fbb03b 77.09%
+      );
+    }
+
+    .modeContainer {
+      min-height: 7rem;
+    }
+
+    .childContainer {
+      position: relative;
+      margin-bottom: 0.5rem;
+    }
+
+    #editAnchorButton {
+      display: flex;
+      flex-direction: column;
+      position: absolute;
+      top: 0;
+      bottom: 0;
+      right: 0;
+      padding: 0.3rem;
+      margin: 0;
+      width: 2rem;
+      background: transparent;
+      border: none;
+    }
+
+    #registerButton {
+      margin: 3rem 0;
+    }
+
+    .hostName {
+      font-size: 1rem;
+      font-weight: 400;
+    }
+
+    .sectionTitle {
+      margin: 1rem 0 0.5rem 0;
+      font-size: 1rem;
+      font-weight: normal;
+      color: black;
+    }
+
+    .buttonContainer {
+      margin: 0.5rem 0;
+    }
+
+    hr {
+      border: none;
+      border-top-color: var(--grey-050);
+      border-top-style: solid;
+      margin: 1rem 0;
+    }
+  </style>
+  <div class="container">
+    <h1>Authorize Authentication</h1>
+    <h2 class="sectionTitle">Application URL</h2>
+    <div class="highlightBox hostName">${hostName}</div>
+    <h2 class="sectionTitle">Identity Anchor</h2>
+    <div class="modeContainer">
+      <div class="childContainer">
+        <input
+          class="anchorText"
+          type="text"
+          id="userNumberInput"
+          placeholder="Enter anchor"
+          value="${userNumber !== undefined ? userNumber?.toString() : ""}"
```

Comment:
I'm curious, why do you need `?.` and `toString()`? I would assume that ts infers `bigint` (and not `bigint | undefined`) and that `.toString()` would be implicit

Reply:
You're right. I think I had to be more explicit in some other context and this is just a left-over artifact of that -> decluttered.